### PR TITLE
Member notifications

### PR DESCRIPTION
**Pull request by szintle**

### CODE REVIEW

File: SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs [hunk 1]
```diff
@@ -61,9 +61,9 @@ public async Task<IActionResult> ddCompetition([FromBody]Competition comp)
             {
                 return BadRequest(ModelState);
             }
-            await _context.AddAsync(comp);
-            await _context.SaveChangesAsync();
-            new HelperClass().Notification(_context, comp);
+            //await _context.AddAsync(comp);
```
Comment (ash-rabbit) [inline SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: did you forget to remove this?

File: SkietbaanBE/SkietbaanBE/Controllers/NotificationController.cs [hunk 2]
```diff
@@ -19,46 +19,41 @@ public NotificationController(ModelsContext context)
...
-        public IEnumerable<Notifications> Get()
+        [HttpGet("{id}", Name = "GetNotificationById")]
+        public async Task<Notifications> GetNotificationById(int id)
         {
-            var notificationQuery = from noti in _context.Notifications
-                                   where noti.IsRead == false
-                                   select noti;
-            List<Notifications> notificationList = notificationQuery.ToList<Notifications>();
-            return notificationList;
+            var notification = _context.Notifications.SingleOrDefault(bright => bright.Id == id);
```
Comment (ash-rabbit) [inline SkietbaanBE/SkietbaanBE/Controllers/NotificationController.cs]: variable name is bright? anyways if it fails to fetch anything what happens?

File: SkietbaanBE/SkietbaanBE/Controllers/NotificationController.cs [hunk 3]
```diff
@@ -19,46 +19,41 @@ public NotificationController(ModelsContext context)
...
-        }
-        
-        // DELETE: api/ApiWithActions/5
         [HttpDelete("{id}")]
-        public void Delete(int id)
+
+        public async Task DeleteAsync(int id)
         {
+            var notification = await GetNotificationById(id);
+           _context.Notifications.Remove(notification);
```
Comment (ash-rabbit) [inline SkietbaanBE/SkietbaanBE/Controllers/NotificationController.cs]: Same question for this one what if connection to db fails

File: SkietbaanBE/SkietbaanBE/Controllers/UserController.cs [hunk 4]
```diff
@@ -87,8 +89,15 @@ public async Task<ActionResult> AddUser([FromBody] User user)
                     userCompStat.User = user;
                     userCompStat.Competition = competitions.ElementAt(i);
                     userCompStat.Best = 0;
-                    userCompStats.Add(userCompStat);
+                   // userCompStat.CompScore = 0;
```
Comment (ash-rabbit) [inline SkietbaanBE/SkietbaanBE/Controllers/UserController.cs]: I think you forgot the gc by mistake

File: SkietbaanBE/SkietbaanBE/Helper/NotificationMessages.cs [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    public class NotificationMessages
+    {
+        public NotificationMessages()
+        {
+
+        }
+
+        public void ConfirmationNotification(ModelsContext _context, User user)
+        {
+            Notifications notification = new Notifications
```
Comment (ash-rabbit) [inline SkietbaanBE/SkietbaanBE/Helper/NotificationMessages.cs]: this can be var

File: SkietbaanBE/SkietbaanBE/Helper/NotificationMessages.cs [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+            {
+                User = user,
+                IsRead = false,
+                typeOfNotification = "Confirmation",
+                NotificationMessage = "Welcome to the club " + user.Username + ". You are now a member."
+            };
+
+            string message = "";
```
Comment (GeromeSchutte) [inline SkietbaanBE/SkietbaanBE/Helper/NotificationMessages.cs]: string.empty is best practice

File: SkietbaanBE/SkietbaanBE/Properties/launchSettings.json [hunk 7]
```diff
@@ -0,0 +1,30 @@
+{
+  "iisSettings": {
+    "windowsAuthentication": false,
+    "anonymousAuthentication": true,
+    "iisExpress": {
+      "applicationUrl": "http://localhost:63474/",
```
Comment (ash-rabbit) [inline SkietbaanBE/SkietbaanBE/Properties/launchSettings.json]: Mistakenly you left the local URL

File: SkietbaanBE/SkietbaanBE/Properties/launchSettings.json [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+      }
+    },
+    "SkietbaanBE": {
+      "commandName": "Project",
+      "launchBrowser": true,
+      "launchUrl": "api/Test",
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Development"
+      },
+      "applicationUrl": "http://localhost:63475/"
```
Comment (ash-rabbit) [inline SkietbaanBE/SkietbaanBE/Properties/launchSettings.json]: here as well

File: SkietbaanBE/SkietbaanBE/Helper/NotificationMessages.cs [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+        public void CompetitionNotification(ModelsContext _context, Competition comp)
+        {
+            var userlist = _context.Users.ToList();
+            foreach (var user in userlist)
+            {
+                Notifications notification = new Notifications();
+                notification.User = user;
+                notification.IsRead = false;
+                notification.typeOfNotification = "Competition";
+                notification.NotificationMessage = "Hello valued member. Please note there is a new competition that has been created: " + comp.Name;
```
Comment (GeromeSchutte) [inline SkietbaanBE/SkietbaanBE/Helper/NotificationMessages.cs]: Consider using string interpolation

File: SkietbaanBE/SkietbaanBE/Models/Notifications.cs [hunk 10]
```diff
@@ -13,8 +13,8 @@ public class Notifications
         [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
         public int Id { get; set; }
         public bool IsRead { get; set; }
-        public string NotificationsHeading { get; set; }
-        public string NotificationContent { get; set; }
+        public string typeOfNotification { get; set; }
```
Comment (GeromeSchutte) [inline SkietbaanBE/SkietbaanBE/Models/Notifications.cs]: watch your casing consistency

File: SkietbaanBE/SkietbaanBE/Controllers/NotificationController.cs (hunk 2 above)
Comment (szintle) [reply SkietbaanBE/SkietbaanBE/Controllers/NotificationController.cs]: Returns NotFound

File: SkietbaanBE/SkietbaanBE/Properties/launchSettings.json [hunk 11]
```diff
@@ -24,7 +24,7 @@
       "environmentVariables": {
         "ASPNETCORE_ENVIRONMENT": "Development"
       },
-      "applicationUrl": "http://localhost:63475/"
+      "applicationUrl": ""
```
Comment (GeromeSchutte) [inline SkietbaanBE/SkietbaanBE/Properties/launchSettings.json]: what will happen if a person clones the project, installs their nuget packages and tries to run it? Will leaving this blank not cause the run to fail?

Comment (szintle) [reply SkietbaanBE/SkietbaanBE/Properties/launchSettings.json]: It will. How should I go about it?

Comment (szintle) [reply SkietbaanBE/SkietbaanBE/Properties/launchSettings.json]: Fixed the problem

**Pull request merged**